Change `maxParallelSession` type to `Option<u8>`

div-seungha opened the pull request:
Closes: #382

sehkone commented:
@div-seungha,

Since the corresponding review-database has been updated, I think this PR should proceed. Could you do that?

div-seungha commented:
> @div-seungha,
> 
> Since the corresponding review-database has been updated, I think this PR should proceed. Could you do that?

I have rebased on the latest main branch and replaced the previously used `and_then`and `try_from` with `map` in the relevant code. The new commit has been pushed.

sophie-cluml commented on CHANGELOG.md:
```diff
@@ -58,6 +58,10 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 - Fixed an issue where configuration conversion failures were silently ignored,
   leading to incorrect None handling.
+- Changed `maxParallelSession` in `Account` to `Option<i32>.` Previously,
+  `maxParallelSession` was returned as a `Option<StringNumber>` type. However,
```
- I think we need to mention that this changelog entry is related to GraphQL API. 
- Is `maxParallelSession` in `Account` changed to `Option<i32>.` ?
- I think it would be beneficial to clarify that the field is within the u8 range.

div-seungha replied:
I have modified the changelog to contain the meaning more clearly.

sehkone commented:
In addition to mentioning the range in the Changelog, GraphQL documentation should be included accordingly.

sophie-cluml commented on CHANGELOG.md:
```diff
@@ -58,6 +58,12 @@ Versioning](https://semver.org/spec/v2.0.0.html).
   set to an empty string were not saved to the `config` in the database.
 - Fixed an issue where configuration conversion failures were silently ignored,
   leading to incorrect None handling.
+- The `maxParallelSession` field in `Account` has been updated to `Option<i32>`,
+  replacing the previous `Option<StringNumber>` type. In the review-database,
+  the `max_parallel_sessions` type has been changed from `Option<u32>` to
+  `Option<u8>`, with the related code updated accordingly. Since GraphQL does
+  not support unsigned integers, the type has been replaced with the signed type
+  `Option<i32>`, which can accommodate the `Option<u8>` values.
```
유저 관점에서, 유저가 알아야 하는 정보를 적어주세요.

sophie-cluml commented on CHANGELOG.md:
```diff
@@ -58,6 +58,12 @@ Versioning](https://semver.org/spec/v2.0.0.html).
   set to an empty string were not saved to the `config` in the database.
 - Fixed an issue where configuration conversion failures were silently ignored,
   leading to incorrect None handling.
+- Updated the `maxParallelSessions` field in `Account` to Option<i32>, replacing
+  the previous `Option<StringNumber>` type. This change aligns with the update
+  in the review-database, where the `max_parallel_sessions` type was changed
+  from `Option<u32>` to `Option<u8>`. Since GraphQL does not support unsigned
+  integers, `Option<i32>` was chosen to ensure compatibility while accommodating
+  the `Option<u8>` values.
```
I think a quick sync meeting for this might help us, @div-seungha would be up for it for a second?

div-seungha replied:
I revised the commit considering we talked about in a separate communication channel.

sophie-cluml commented on CHANGELOG.md:
```diff
@@ -46,6 +46,13 @@ Versioning](https://semver.org/spec/v2.0.0.html).
+  - The `account` and related queries such as `accountList` now return
+    `maxParallelSessions` as an `Int` within the range of `u8`.
+  - The `insertAccount` and `updateAccount` GraphQL APIs remain unchanged in
+    their interfaces but now only accept `maxParallelSessions` parameters within
```
Does `updateAccount` have `maxParallelSessions` parameter?

div-seungha replied:
> but now only accept parameters related to max parallel sessions within the range of `u8`.

No, It was my mistake. I have just modified the sentence.
